test-case-reporting/types/test-case-reporting.d.ts

```diff
@@ -18,65 +18,106 @@ declare module 'test-case-reporting' {
   /**
    * Travis job types for which test results may be reported.
    */
-  export type JobType = 'unit' | 'integration';
+  export type TestSuiteType = 'unit' | 'integration';
 
-  export type TestStatus = 'pass' | 'fail' | 'skip' | 'error';
+  export type TestStatus = 'PASS' | 'FAIL' | 'SKIP' | 'ERROR';
 
   /** A build on Travis. */
   export interface Build {
-    username: string,
-    commitHash: string,
-    prNumber: number,
-    startedAt: Date,
+    username: string;
+    commitHash: string;
+    prNumber: number;
+    startedAt: Date;
 
     // The list of jobs we know are contained in the build.
     // When we create a build, we fill this with its jobs, but when
     // we get a build from a server, the jobs list may not be populated
     // if the jobs are not needed.
-    jobs: Array<Job>,
+    jobs: Array<Job>;
   }
 
   /** A job within a Travis build. */
   export interface Job {
-    jobNumber: string,
-    type: JobType,
-    startedAt: Date,
+    jobNumber: string;
+    testSuiteType: TestSuiteType;
+    startedAt: Date;
 
     // This list is treated similarly to the `jobs` array of the `Build` type.
     // Read the comment on `jobs` for info.
-    testRuns: Array<TestRun>,
+    testRuns: Array<TestRun>;
   }
 
   /** A single kind of test case, one `it` or `test` block. */
   export interface TestCase {
-    name: string,
-    createdAt: Date,
+    name: string;
+    createdAt: Date;
   }
 
   /** An instance of a test being run, with results. */
   export interface TestRun {
-    testCase: TestCase,
-    status: TestStatus,
-    timestamp: Date,
-    durationMs: number,
+    testCase: TestCase;
+    status: TestStatus;
+    timestamp: Date;
+    durationMs: number;
+  }
+
+  // Types in the DB namespace interface with the database which
+  // uses snake_case instead of camelCase.
+  /* eslint @typescript-eslint/camelcase: "off" */
+  namespace DB {
+    export interface Build {
+      // `id` is nullable because is not set when uploading, it is only set
+      // when we get the build from the database.
+      // It is not nullable in the database.
+      id?: number;
+      commit_hash: string;
+      pull_request_number: number;
+      started_at: number;
+    }
+
+    export interface Job {
+      // See comment under `DB.Build.id`
+      id?: number;
+      buld_id: number;
+      job_number: string;
+      test_suite_type: string; // e.g. unit, integration, etc.
```

Comment:
Should this be type `TestSuiteType`?

Reply:
Oh, good point.

Instead of turning this into `TestSuiteType`, I think I'll turn the other field into a string and remove the enum to follow  https://github.com/ampproject/amp-github-apps/pull/896#discussion_r441030655. Not sure, though -- what do you think?

Reply:
The comment was about removing the _database_ enum type. In typescript, however, we still want the type-checking/validation. I agree with making this `TestSuiteType`

Reply:
Done.